### PR TITLE
Enhance Tax Calculator with shipping items taxation logic

### PR DESCRIPTION
**Pull request by ParameshOddepally**
This commit enhances the Tax calculator sample and adds the required logic to tax shipping items in cart request.

### CODE REVIEW

**Review by shivani-khandelwal — APPROVED** [review]

**Review by swati681 — APPROVED** [review]
Have a work item to add tests for this calculator. Extensibility team has enforced each checkin to have 75% + test coverage. Refer pricing calculator tests in the same repo.

File: commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls [hunk 1]
```diff
@@ -146,10 +164,40 @@ public class TaxCartCalculatorSample extends CartExtension.TaxCartCalculator {
...
               );
-              adj.getCartTaxes().add(promoTax);
+              if (adj != null) {
+                adj.getCartTaxes().add(promoTax);
+              }
             }
         }
       }
+
+      // If there are shipping items, add tax for them as well
```
Comment (snilakandan13) [inline commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls:174]: How are you planning on testing this? Would you be modifying the Apex test class?

Comment (snilakandan13) [reply commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls:174]: Looks like you may need to modify the test cases to test this scenario

Comment (ParameshOddepally) [reply commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls:174]: We do not have any tests for this currently. I tested the use cases manually on steam org. We plan to raise a work item to cover the tests.

Comment (snilakandan13) [reply commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls:174]: Please add fTests as appropriate and Unit tests for this apex class.

**Review by snilakandan13 — APPROVED** [review]

**Pull request merged**